dev/integration_tests/web_e2e_tests/test_driver/text_editing_integration.dart

```diff
@@ -15,10 +15,31 @@ import 'package:web_e2e_tests/text_editing_main.dart' as app;
 void main() {
   IntegrationTestWidgetsFlutterBinding.ensureInitialized();
 
+  ShadowRoot? shadowRoot;
+  bool shadowRootInitialized = false;
+
+  /// Attempts to detect if the engine is rendering with a shadowRoot in the
+  /// glass-pane, or not.
+  void initializeShadowRoot() {
+    if (!shadowRootInitialized) {
```

Comment:
Remove the bool check here, and move the shadowRoot selector to the findElements function directly? It's not a huge performance hit, and that way we don't have to call initializeShadowRoot all the time.